x/market/types/types_test.go

```diff
@@ -0,0 +1,120 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/ovrclk/akash/sdl"
+	"github.com/ovrclk/akash/testutil"
+	"github.com/ovrclk/akash/x/market/types"
+	"github.com/pkg/errors"
+)
+
+func TestOrderValidateCanMatch(t *testing.T) {
+	baseHeight := int64(10000)
+	boostHeight := int64(50)
+
+	type testCase struct {
+		desc   string
+		order  types.Order
+		height int64
+		expErr error
+	}
+	tests := []testCase{
+		{
+			desc: "zero values",
+			order: types.Order{
+				OrderID: testutil.OrderID(t),
+				StartAt: baseHeight,
+			},
+			height: baseHeight,
+			expErr: types.ErrOrderClosed,
+		},
+		{
+			desc: "zero value CloseAt",
+			order: types.Order{
+				OrderID: testutil.OrderID(t),
+				State:   types.OrderOpen,
+				StartAt: baseHeight,
+				// CloseAt is 0
+			},
+			height: baseHeight + sdl.DefaultOrderBiddingDuration + boostHeight,
+			expErr: types.ErrOrderDurationExceeded,
+		},
+		{
+			desc: "zero value CloseAt within default height",
+			order: types.Order{
+				OrderID: testutil.OrderID(t),
+				State:   types.OrderOpen,
+				StartAt: baseHeight,
+				// CloseAt is 0
+			},
+			height: baseHeight + boostHeight + boostHeight,
+			expErr: nil,
+		},
+		{
+			desc: "open order returns no error",
+			order: types.Order{
+				OrderID: testutil.OrderID(t),
+				State:   types.OrderOpen,
+				StartAt: baseHeight,
+				CloseAt: baseHeight + boostHeight,
+			},
+			height: baseHeight + boostHeight - int64(10),
+			expErr: nil,
+		},
+		{
+			desc: "block height beyond CloseAt",
+			order: types.Order{
+				OrderID: testutil.OrderID(t),
+				State:   types.OrderOpen,
+				StartAt: baseHeight,
+				CloseAt: baseHeight + boostHeight,
+			},
+			height: baseHeight + boostHeight + boostHeight,
+			expErr: types.ErrOrderDurationExceeded,
+		},
+		{
+			desc: "state errors makes sense",
+			order: types.Order{
+				OrderID: testutil.OrderID(t),
+				State:   types.OrderClosed,
+				StartAt: baseHeight,
+				CloseAt: baseHeight + boostHeight,
+			},
+			height: int64(120),
+			expErr: types.ErrOrderClosed,
+		},
+		{
+			desc: "assert order matched error",
+			order: types.Order{
+				OrderID: testutil.OrderID(t),
+				State:   types.OrderMatched,
+				StartAt: baseHeight,
+				CloseAt: baseHeight + boostHeight,
+			},
+			height: int64(80),
+			expErr: types.ErrOrderMatched,
+		},
+		{
+			desc: "height less than allowed start",
+			order: types.Order{
+
+				OrderID: testutil.OrderID(t),
+				State:   types.OrderOpen,
+				StartAt: baseHeight,
+				CloseAt: baseHeight + boostHeight,
+			},
+			height: int64(90),
+			expErr: types.ErrOrderTooEarly,
+		},
+	}
+	for _, test := range tests {
+		t.Logf("%s", test.desc)
```

Comment:
:hankey:

Reply:
I need to make a regex sublinter for this.. sorry.

Reply:
removed all `t.Logf()` from repository.